docs/specs/admin-api-change-password.md

```diff
@@ -0,0 +1,84 @@
+# Change Password in Admin API
+
+- [Introduction](#introduction)
+- [Use Cases](#use-cases)
+- [Design](#design)
+  - [Change Password](#change-password)
+    - [Error Response](#error-response)
+  - [Force Change Password on Next Login](#force-change-password-on-next-login)
+    - [Error Response](#error-response-1)
+
+## Introduction
+
+This document describes the feature for administrators to change the password of a user through Admin API.
+
+## Use Cases
+
+- An administrator needs to change the password of an existing user.
+- An administrator needs to force a user to change their password.
+- An administrator needs to reset the password of a user who has forgotten their password.
+
+## Design
+
+### Change Password
+
+`resetPassword` mutation is an existing mutation in the Portal Admin API. It will be extended to support following:
+- Send new password to user through email.
+- Force user to change their password on next login.
+- Generate a new password automatically. In this case, new password will be sent to user through email.
+
+```graphql
+type Mutation {
+  # other root fields...
+
+  resetPassword(
+    input: ResetPasswordInput!
+  ): ResetPasswordPayload!
+}
+
+input ResetPasswordInput {
+  userID: ID!
+  # Optional. It was required previously.
+  # If password is not given, then it is generated by Authgear and sent to user through email regardless of `sendPassword` value.
+  password: String
+  # Optional. Default is false. If set to true, then new password is sent to user through email.
+  sendPassword: Boolean
+  # Optional. Default is false. If set to true, then user will be forced to change password on next login.
+  markPasswordAsExpired: Boolean
+}
+```
+
+#### Error Response
+
+|Description|Name|Reason|Info|
+|---|---|---|---|
+|Email identity not found, required if `sendPassword` is set to `true` or password is auto-generated.|`Invalid`|`SendPasswordNoTarget`|-|
```

Comment:
Change the description to 

> If `sendPassword` is true, and the user has no verified email to send the password to

That is, the error will not be thrown if `sendPassword` is false, even if the password is generated.